File: backend/app/schemas/schedule.py
import datetime

from pydantic import BaseModel


class TeacherBrief(BaseModel):
    id: int
    nombre: str


class DayScheduleEntry(BaseModel):
    tramo: int
    asignatura: str | None
    aula: str | None
    titular: TeacherBrief | None   # None si el tramo no tiene profesor asignado
    ausente: bool
    motivo_ausencia: str | None
    sustituto: TeacherBrief | None
    sustitucion_id: int | None
    sustitucion_estado: str | None
    ai_propuesta: bool


class WeekDaySchedule(BaseModel):
    fecha: datetime.date
    dia_semana: int          # 0=Lunes … 4=Viernes
    tramos: list[DayScheduleEntry]

File: frontend/src/api/schedule.ts
import api from './client'

export interface TeacherBrief {
  id: number
  nombre: string
}

export interface DayScheduleEntry {
  tramo: number
  asignatura: string | null
  aula: string | null
  titular: TeacherBrief | null
  ausente: boolean
  motivo_ausencia: string | null
  sustituto: TeacherBrief | null
  sustitucion_id: number | null
  sustitucion_estado: string | null
  ai_propuesta: boolean
}

export interface WeekDaySchedule {
  fecha: string       // YYYY-MM-DD
  dia_semana: number  // 0=Lunes … 4=Viernes
  tramos: DayScheduleEntry[]
}

export const getCursos = async (): Promise<string[]> => {
  const response = await api.get<string[]>('/schedule/cursos')
  return response.data
}

export const getDaySchedule = async (fecha: string, curso: string): Promise<DayScheduleEntry[]> => {
  const response = await api.get<DayScheduleEntry[]>('/schedule/day', {
    params: { fecha, curso },
  })
  return response.data
}

export const getWeekSchedule = async (fecha: string, curso: string): Promise<WeekDaySchedule[]> => {
  const response = await api.get<WeekDaySchedule[]>('/schedule/week', {
    params: { fecha, curso },
  })
  return response.data
}
